From 1dd3372d13da28982fcd96d78c13711f56d11b8c Mon Sep 17 00:00:00 2001
From: Iiro Valkonen <iiro.valkonen@atmel.com>
Date: Fri, 29 Apr 2011 09:37:45 -0400
Subject: [PATCH 24/57] CHROMIUM: INPUT: atmel_mxt_ts - Increase the wait times
 for backup and reset

Increase the sleep times after backup and reset, so that we can be sure they have been completed even when the mXT cycle time is set to maximum value (254ms).

Signed-off-by: Iiro Valkonen <iiro.valkonen@atmel.com>
Patch is in progress of being submitted at http://www.spinics.net/lists/linux-input/msg14593.html
Signed-off-by: Jon Kliegman <kliegs@chromium.org>

R=olofj@chromium.org, msb@chromium.org
BUG=chromium-os:13514
TEST=Built kernel
  Validated kernel boots fine
  Validated in conjunction with other CLs in bug that the driver works

Cherrypicking forward to 3.0 branch from 2.6.38
(cherry picked from commit 1ebffd5d7df2202b20eee3c906096ff7f1d6c56a)

Cherrypicking forward to 2.6.38 branch from http://codereview.chromium.org/6711073
(cherry picked from commit 4dc5fe662e1352e12181ba4cfeb0f8333486c882)

Review URL: http://codereview.chromium.org/6893098
Patch from Iiro Valkonen <iiro.valkonen@atmel.com>.

Change-Id: I487c75fd05d5eb7b6223b5a3b79d34076d096726
Signed-off-by: Yufeng Shen <miletus@chromium.org>
Reviewed-on: http://gerrit.chromium.org/gerrit/8773
---
 drivers/input/touchscreen/atmel_mxt_ts.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/input/touchscreen/atmel_mxt_ts.c b/drivers/input/touchscreen/atmel_mxt_ts.c
index 4d67eeb..2f22ae1 100644
--- a/drivers/input/touchscreen/atmel_mxt_ts.c
+++ b/drivers/input/touchscreen/atmel_mxt_ts.c
@@ -195,8 +195,8 @@
 /* Define for MXT_GEN_COMMAND_T6 */
 #define MXT_BOOT_VALUE		0xa5
 #define MXT_BACKUP_VALUE	0x55
-#define MXT_BACKUP_TIME		25	/* msec */
-#define MXT_RESET_TIME		65	/* msec */
+#define MXT_BACKUP_TIME		270	/* msec */
+#define MXT_RESET_TIME		350	/* msec */
 #define MXT_CAL_TIME		25	/* msec */
 
 #define MXT_FWRESET_TIME	500	/* msec */
-- 
1.8.1.4

